packages/jest-types/__typetests__/expect.test.ts

```diff
@@ -221,6 +221,11 @@ expectType<void>(
   ),
 );
 
+expectType<void>(
+  expect({name: 'someName', age: 12}).toBe({name: 'someOtherName', age: 13}),
+);
+expectError(expect({name: 'someName', age: 12}).toBe({name: 'someOtherName'}));
```

Comment:
Hm.. But both of these will fail at run time, right?

Reply:
yep, both fail at run time
first one passes type checks
second one fails type checks